### PR TITLE
017 pagination

### PR DESCRIPTION
**Pull request by Adnanhoq**

### CODE REVIEW

**Review by sdagn1 — APPROVED** [review]

File: test/unit/services/JobRoleServiceTests.ts [hunk 1]
```diff
@@ -43,22 +54,23 @@ describe('JobRoleService', function () {
...
-        
+        const results = await getJobRoles(1,10);
+        const { jobRoles, pagination } = results;
+
+        jobRoles[0].closingDate = new Date(jobRoles[0].closingDate);
 
-        results[0].closingDate = new Date(results[0].closingDate);
-        expect(results[0]).to.deep.equal(jobRoleResponse);
+        expect(results).to.deep.equal(jobRoleResponseWrapper);
       })
```
Comment (JennicaMistry) [inline test/unit/services/JobRoleServiceTests.ts:68]: You need to have tests that have a different page number and different limits for all scenarios - e.g the happy path and the two exceptions.

File: views/jobRoles.njk [hunk 2]
```diff
@@ -13,9 +13,26 @@
         <div class="container">
 
             <div class="mainBody">
-                <h1>Job Roles List</h1>
+                <h1 class = "m-0 p-0">Job Roles List</h1>
+                <div class = "d-flex flex-row justify-content-between m-0 p-0">
```
Comment (JennicaMistry) [inline views/jobRoles.njk:17]: I want a line gap between the Job Roles List title and the pagination total at the top of the page and the drop down should be inline with the pagination total after this change.

File: views/jobRoles.njk [hunk 3]
```diff
@@ -29,6 +46,40 @@
...
+                    </div>
+                </div>
+
+                <nav class = "d-flex flex-row justify-content-center mb-5 pb-5">
+                    <ul class="pagination">
+                        {% if pagination.page > 1 %}
+                            <li class="page-item">
+                                <a class="page-link" href="?page={{ pagination.page - 1 }}&limit={{ pagination.limit }}">Previous</a>
+                            </li>
+                        {% endif %}
```
Comment (JennicaMistry) [inline views/jobRoles.njk:70]: As per the ticket the UI looks odd when you have one card at the end of the page but still have content on another page. There are two ways to fix this. Either you can have 5 cards, but risk smaller text, or 1 card per row. Pick one.

File: views/jobRoles.njk [hunk 4]
```diff
@@ -13,9 +13,26 @@
         <div class="container">
 
             <div class="mainBody">
-                <h1>Job Roles List</h1>
+                <h1 class = "m-0 p-0">Job Roles List</h1>
```
Comment (JennicaMistry) [inline views/jobRoles.njk:16]: I want the Job Roles List title in the middle of the page.

**Review by JennicaMistry — CHANGES_REQUESTED** [review]

File: views/jobRoles.njk (hunk 4 above)
Comment (Adnanhoq) [reply views/jobRoles.njk:16]: Done

File: views/jobRoles.njk (hunk 2 above)
Comment (Adnanhoq) [reply views/jobRoles.njk:17]: Added

File: views/jobRoles.njk (hunk 3 above)
Comment (Adnanhoq) [reply views/jobRoles.njk:70]: Overruled by you guys

File: test/unit/services/JobRoleServiceTests.ts (hunk 1 above)
Comment (Adnanhoq) [reply test/unit/services/JobRoleServiceTests.ts:68]: Added

**Review by JennicaMistry — DISMISSED** [review]

**Review by JennicaMistry — APPROVED** [review]

**Review by sdagn1 — APPROVED** [review]

**Pull request merged**